stdlib/deplinker/elf.py

```diff
@@ -0,0 +1,206 @@
+
+        packages[package_id.short_name()].run_dependencies.update(requirements)
+
+
+def _solve_remotely(dependency) -> Optional[str]:
+    config = core.config.get_config()
+
+    if config.get('repositories') is None:
+        return None
+
+    # Try all repositories from top to bottom
+    for repository in config['repositories']:
+        try:
+            url = core.config.get_config()['repositories'][repository]['url']
+            r = requests.get(
+                url=f'{url}/api/search?q={urllib.parse.quote(dependency)}&search_by=content&exact_match=true',
+            )
+
+            if r.status_code == 200:
+                results = r.json()
+
+                if len(results) == 1:
+                    result = results[0]
+                    if not result['all_versions']:
+                        stdlib.log.elog(f"\"{repository}\" contains a single package with file \"{dependency}\" but for all versions")
+                    else:
+                        stdlib.log.slog(f"\"{repository}\" contains a single package with file \"{dependency}\"!")
+                        return result['name']
+                elif len(results) > 1:
+                    stdlib.log.elog(f"\"{repository}\" contains more than one package with file \"{dependency}\"")
+                else:
+                    stdlib.log.elog(f"\"{repository}\" doesn't contain any package with file \"{dependency}\"")
+            elif r.status_code == 404:
+                stdlib.log.elog(f"\"{repository}\" doesn't contain a package with file \"{dependency}\"")
+            else:
+                raise RuntimeError("Repository returned an unknown status code")
+        except:
+            stdlib.log.elog(f"An unknown error occured when fetching \"{repository}\" (is the link dead?), skipping...")
+
+    return None
+
+
+def _fetch_elf_dependencies(package, elf_path) -> [str]:
+    deps = []
+
+    with stdlib.pushd(package.wrap_cache):
+        try:
+            with open(elf_path, 'rb') as file:
+                elf = ELFFile(file)
+                dyn = elf.get_section_by_name(".dynamic")
+                if dyn is not None:
+                    for tag in dyn.iter_tags():
+                        if tag.entry.d_tag == 'DT_NEEDED':
+                            deps += [tag.needed]
+        except:
+            pass  # Ignore invalid ELFs (dangling/text files in bin/ or lib/)
+
+    return deps
+
+
+def _find_elfs(package, search_patterns) -> [str]:
+    files = []
+
+    with stdlib.pushd(package.wrap_cache):
+        for search_pattern in search_patterns:
+            for rglob in braceexpand.braceexpand(search_pattern):  # Expand braces
+                for rpath in glob.glob(rglob):  # Expand globbing
+
+                    # We want to retain ELFs only
+                    #
+                    # Unfortunately, I couldn't find a better way but to try to open the file
```

Comment:
That still seems not that far from Python's "better ask for forgiveness than permission" philosophy :p

Reply:
As a mostly Rust/C developer, I feel quite sad and disturbed about that :(